### PR TITLE
User config trough configurator. Some small bug fixes

### PR DESCRIPTION
**Pull request by carlhampuswall**
User config trough configurator, new protobuf for settings.
Fixes led issues and adds delay for fade in/out effects.
Blend between colors lighthouse etc.

### CODE REVIEW

File: firmware/src/led_ring/led_ring_task.cpp [hunk 1]
```diff
@@ -261,40 +338,56 @@ void LedRingTask::run()
         // This is useful for checking the queue and responding to events without delaying program execution.
         if (xQueueReceive(render_effect_queue_, &effect_settings, 0) == pdTRUE)
         {
-            // Sync reaction to receiving an effect.
-        }
+            if (effect_settings.effect_id == old_effect_settings.effect_id) // PREVENTS LED FLICKERING (only render new effects)
+            {
+                if (effect_settings.effect_id == 7 && effect_settings.effect_brightness == old_effect_settings.effect_brightness)
```
Comment (nicolabortignon) [inline firmware/src/led_ring/led_ring_task.cpp:343]: Couple of thoughts, for good refactoring opportunity. 

When yo uhave situation like effect_setting.effect_id == 7 in a condition, you really want to go back and change the 7 into a verbose constant. So someone doesn't need to go back and look up what 7 means. 
The if becomes effect_settings.effect_id == EFFECT_LIGHTHOUSE for instance. where EFFECT_LIGHTHOUSE = 7. 

A more complex thought:

Here you have a situation with nested If(s). Even more, there is a switch on effect just after. Always be worried when you see something like this.
It means there  is poor encapsulation somewhere. In this case you can see how the Run method, and the specific effect 7, are sharing the role in determine when the effect is over. You want to figure out a way to move that completley inside the effect itself. And, leave the method run to just execute whatever effect (without having to know the logic of each effect)

**Review by nicolabortignon — APPROVED** [review]
Some comments, and probably the opportunity to do some  small changes now (vs some more heavy refactoring that we'll do later on)

File: firmware/src/configuration.h [hunk 2]
```diff
@@ -67,6 +68,31 @@ struct OSConfiguration
     OSMode mode = ONBOARDING;
 };
 
+static const SETTINGS_Settings default_settings =
+    {
+        .has_screen = true,
```
[the remark this answers is not in the document]
Comment (carlhampuswall) [reply firmware/src/configuration.h:73]: This is automatically generated from PROTOBUF.

**Comment by nicolabortignon** [comment]
Ah! that's annoying :(

File: firmware/src/led_ring/led_ring_task.cpp [hunk 3]
```diff
@@ -97,36 +97,63 @@ void LedRingTask::renderTrailEffect()
 
 void LedRingTask::renderEffectLightHouse()
 {
-    const uint32_t frame_ms = 100;
+    CRGB targetColor = CRGB(effect_settings.effect_main_color);
+    CRGB accentColor = CRGB(effect_settings.effect_accent_color);
```
[the remark this answers is not in the document]
Comment (carlhampuswall) [reply firmware/src/led_ring/led_ring_task.cpp:101]: If im not mistaken in this context accent  color is used as the previous color (other led colors) and main color is the beacons color. But im fine with changing to a more correct naming approach, just utilized the current implementation without trying to change to much.

File: firmware/src/led_ring/led_ring_task.cpp [hunk 4]
```diff
@@ -165,25 +192,25 @@ void LedRingTask::renderFadeInEffect()
...
-                               (b * ledsBrightness[i]) / FULL_BRIGHTNESS);
                 ledsBrightness[i]++;
             }
+            leds[i].setColorCode(effect_settings.effect_main_color);
         }
 
         // Show the LEDs with the updated brightness level
+        FastLED.setBrightness(ledsBrightness[0]);
         FastLED.show();
+        delay(50);
```
[the remark this answers is not in the document]
Comment (carlhampuswall) [reply firmware/src/led_ring/led_ring_task.cpp:206]: Same as my comment above

File: firmware/src/led_ring/led_ring_task.cpp (hunk 1 above)
Comment (carlhampuswall) [reply firmware/src/led_ring/led_ring_task.cpp:343]: Just utilized @brushknight original code or whoever wrote it from the start. (Done like u suggest in my other code using enums and switch statements etc.)

But ill refactor into this. 👍

File: firmware/src/led_ring/led_ring_task.cpp [hunk 5]
```diff
@@ -97,36 +97,63 @@ void LedRingTask::renderTrailEffect()
...
+                uint8_t scaledBright = scale8(ledsBrightness[i], maxABright);
+
+                leds[i].r = scale8(accentColor.r, scaledBright);
+                leds[i].g = scale8(accentColor.g, scaledBright);
+                leds[i].b = scale8(accentColor.b, scaledBright);
+            }
         }
 
         FastLED.show();
+        delay(10);
```
[the remark this answers is not in the document]
Comment (carlhampuswall) [reply firmware/src/led_ring/led_ring_task.cpp:154]: Should be done by checking millis() and storing that etc will refactor.

File: firmware/src/led_ring/led_ring_task.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (carlhampuswall) [reply firmware/src/led_ring/led_ring_task.cpp:101]: Will keep it as is for now, as it kinda is accent and main color. Main color is the color we are transitioning the led ring to (to one led the beacon led) accent color is what we are transitioning from.

**Pull request merged**